### ISSUE TITLE
@babel/plugin-transform-runtime is not compatible with @babel/plugin-transform-proto-to-assign

### ISSUE THREAD

**Issue by 851201344** [babel/babel#11997]
## Bug Report

- [ ] I would like to work on a fix!

**Current behavior**
I have a problem with a similar premise https://github.com/babel/babel/issues/6617 , https://github.com/babel/babel/issues/116

I solved this problem by using the **@babel/plugin-transform-proto-to-assign** plugin
[https://babeljs.io/docs/en/caveats#classes-10-and-below](url)

but

If I use the @babel/ plugin-Transform-Runtime plug-in, I won't be able to run under IE10/9.

**Input Code**
```js
    class Base {
      static getInstance() {
        console.log(111);
      }
      constructor({ name }) {
        this.name = name;
      }
    }
    class Child extends Base {
      constructor() {
        super({ name: "babel" });
        this.age = 12;
      }
    }
    Child.getInstance();
    var child = new Child()
    console.log(child.age);
    console.log(child.name);
```
```js
module.exports = function (api) {
    api.cache(true);
    return {
        "presets": [
            [
                "@babel/preset-env",
                {
                    'modules': false,
                    'useBuiltIns': 'entry', //entry  usage
                    'corejs': 3,
                    "loose": true,
                    'targets': {
                        'browsers': ['ie >= 8']
                    }
                }
            ]
        ],
        "plugins": [
            "@babel/plugin-transform-runtime",
            "@babel/plugin-proposal-class-properties",
            "@babel/plugin-transform-modules-commonjs", // Cannot assign to read only property 'exports' of object '#<Object>'
            "@babel/plugin-syntax-export-default-from",
            "@babel/plugin-proposal-export-default-from",
            "@babel/plugin-transform-proto-to-assign",
        ],
    };
}
```
 

**Environment**

```
-Babel version(s): [ "@babel/core": "~7.11.1",
    "@babel/plugin-proposal-class-properties": "~7.10.4",
    "@babel/plugin-proposal-export-default-from": "~7.10.4",
    "@babel/plugin-syntax-export-default-from": "~7.10.4",
    "@babel/plugin-transform-classes": "~7.10.4",
    "@babel/plugin-transform-modules-commonjs": "~7.10.4",
    "@babel/plugin-transform-proto-to-assign": "7.10.5",
    "@babel/plugin-transform-runtime": "~7.11.0",
    "@babel/preset-env": "~7.11.0",
    "@babel/runtime": "~7.11.2",
    "@babel/runtime-corejs3": "~7.11.2",
    "webpack": "^4.41.2",]
- OS: [Windows 10]
- How you are using Babel: [`loader`]
- Node/npm version: [e.g. Node 12.14/npm 6.13]
```

**Comment by JMarkoski** [comment]
The problem is the following (I don't know of a workaround to fix it):

These two are babel helpers that babel adds to your code when transpiling, for it to be able to run on older browsers that don't have `extends` support for classes, and that here means Internet Explorer (it doesn't have support for any class features as can be seen from here [https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes):

```
function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
```

As you can see from the code above, `_inherits` uses the `_setPrototypeOf` helper function. Now, take a look at this code here from the `_setPrototypeOf` function:

`o.__proto__ = p; `

This is the code you want to get rid of if you want IE10 and IE9 support. And that is what the `@babel/plugin-transform-proto-to-assign` does: It  transforms the helper function `_setPrototypeOf` to the one below, also shown together with the `_inherits` function and the newly added `_defaults` function that is used in the place of the `_proto_` code:

```
function _defaults(obj, defaults) { var keys = Object.getOwnPropertyNames(defaults); for (var i = 0; i < keys.length; i++) { var key = keys[i]; var value = Object.getOwnPropertyDescriptor(defaults, key); if (value && value.configurable && obj[key] === undefined) { Object.defineProperty(obj, key, value); } } return obj; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { _defaults(o, p); return o; }; return _setPrototypeOf(o, p); }
```

This is good if you only use the `@babel/plugin-transform-proto-to-assign` plugin, and don't touch the `@babel/plugin-transform-runtime` plugin. The problem with including the second plugin is that it transforms the helper functions in this format:

`var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));`

Yes, there is only one helper function, `_inherits`, and the other one you saw from above, `_setPrototypeOf`, is gone. That is because `_inherits` uses it internally. Now, the `@babel/plugin-transform-proto-to-assign` plugin doesn't see any use of the `__proto__` thing in the code, and thus does nothing. The `_setPrototypeOf` function uses `__proto__` but the plugin can't see it, so your code will fail in IE10 and IE9. Now you may think, I may circumvent this by running the `@babel/plugin-transform-proto-to-assign` plugin first (by including it first in the plugins array, see plugin ordering: [https://babeljs.io/docs/en/plugins#plugin-ordering](https://babeljs.io/docs/en/plugins#plugin-ordering), but that won't help either, because the `_defaults` is used in `_setPrototypeOf` and `_setPrototypeOf` is used internally in `_inherits` and your code needs only `_inherits`, so babel will only include that helper. 

To sum up, you somehow need to transpile the `_setPrototypeOf` helper, but apart from this observation, I don't know a workaround that will fix this, someone from the babel team can surely give more insights on this, this is only my limited understanding.

Thanks for reading :)
